tests/test_fixtures.py

```diff
@@ -212,33 +212,57 @@ def test_drop_table(db_testdir):
     '''
     Make sure that we can drop tables and verify they do not exist in the context
     of a test.
+
+    NOTE: For MySQL `DROP TABLE ...` statements "cause an implicit commit after
+    executing. The intent is to handle each such statement in its own special
+    transaction because it cannot be rolled back anyway."
+    https://dev.mysql.com/doc/refman/5.7/en/implicit-commit.html
+
+    So this test is skipped for 'mysql' engines
```

Comment:
What happens if you don't skip this test in MySQL? I would expect an autocommit not to cause a problem here, since the outermost transaction is supposed to listen for nested inner commits and restart transactions when they occur. Maybe there's something about "implicit commits" in MySQL that overrides nested transactions?

I'd like to get to the bottom of this behavior, since it seems pretty restricting to not be able to run any of the statements that are listed on the MySQL docs for implicit commits (`DROP TABLE` and `TRUNCATE TABLE`  are used pretty often in the test suite that inspired this plugin, for example). Again, let me know if you get stuck and I'd be happy to help out with research.

Reply:
The reason I started looking into it was because the tests were failing to begin with, and I was really unaware as to why. I just thought to check that there might be a Transaction violation for DROP TABLE, and the MySQL docs is what I found in my search results.

Quoting from the docs: "The statements listed in this section (and any synonyms for them) implicitly end any transaction active in the current session, **as if you had done a COMMIT before executing the statement.**

Most of these statements also cause an implicit commit after executing. The intent is to handle each such statement in its own special transaction because it cannot be rolled back anyway." 

And the way MySQL does nested transactions in the InnoDB engine is to use `SAVEPOINT`. If you use the `COMMIT` statement, it will commit the top level transaction.

Just to be sure, because I thought I might have misunderstood, I decided to just run a nested transaction manually. And they confirmed the behaviour observed.

That being said, it is reasonable to seek a more confident answer.

These are the statements I ran:
```
CREATE TABLE Test (id int primary key, some_value varchar(20));
SHOW TABLES;
# shows `Test` table is listed
SET autocommit = 0; # Enabled/Disabled does not seem to affect the results

START transaction;
# outer transaction has started

SAVEPOINT innertransaction;
# inner transaction is started -- this is how nested transactions are done in InnoDB

DROP TABLE Test;
SHOW TABLES;
# `Test` table is not listed

ROLLBACK TO innertransaction;
# > ERROR 1305 (42000): SAVEPOINT innertransaction does not exist

# the remaining lines below should have showed `Test` as recovered, but it does not
SHOW TABLES;

ROLLBACK;
```

Reply:
I found a much more concrete reference in the MySQL docs:
https://dev.mysql.com/doc/refman/5.7/en/cannot-roll-back.html

> Some statements cannot be rolled back. In general, these include data definition language (DDL) statements, such as those that create or drop databases, those that create, drop, or alter tables or stored routines.

Reply:
Thanks for doing this research! I'll spend some time with your references to make sure that we're on the same page. If it's true that there's no way to get `DROP TABLE` and `TRUNCATE` to play nicely with MySQL transactions, it'd probably be worth leaving a note on this in the docs.